apollo-converters/pojo/src/test/java/com/apollographql/android/converter/github/IntegrationTest.java

```diff
@@ -0,0 +1,105 @@
+import com.apollographql.android.converter.pojo.ApolloConverterFactory;
+import com.apollographql.android.converter.pojo.CustomTypeAdapter;
+import com.apollographql.android.converter.pojo.OperationRequest;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+import retrofit2.Call;
+import retrofit2.Retrofit;
+import retrofit2.converter.moshi.MoshiConverterFactory;
+import retrofit2.http.Body;
+import retrofit2.http.POST;
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class IntegrationTest {
+
+  private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.US);
+
+  private Service service;
+
+  interface Service {
+    @POST("graphql")
+    Call<Response<UserWithOrganizationsQuery.Data>> heroDetails(@Body OperationRequest<UserWithOrganizationsQuery.Variables> query);
+  }
+
+  @Rule public final MockWebServer server = new MockWebServer();
+
+  @Before public void setUp() {
+    CustomTypeAdapter<Date> dateCustomTypeAdapter = new CustomTypeAdapter<Date>() {
+      @Override public Date decode(String value) {
+        try {
+          return DATE_FORMAT.parse(value);
+        } catch (ParseException e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      @Override public String encode(Date value) {
+        return null;
+      }
+    };
+
+    CustomTypeAdapter<URI> uriCustomTypeAdapter = new CustomTypeAdapter<URI>() {
+      @Override public URI decode(String value) {
+        try {
+          return URI.create(value);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      @Override public String encode(URI value) {
+        return null;
+      }
+    };
+
+    Retrofit retrofit = new Retrofit.Builder()
+        .baseUrl(server.url("/"))
+        .addConverterFactory(new ApolloConverterFactory.Builder()
+            .withResponseFieldMapper(UserWithOrganizationsQuery.Data.class,
+                new UserWithOrganizationsQuery.Data.Mapper(UserWithOrganizationsQueryDataModel.FACTORY))
+            .withCustomTypeAdapter(CustomType.DATETIME, dateCustomTypeAdapter)
+            .withCustomTypeAdapter(CustomType.URI, uriCustomTypeAdapter)
+            .build())
+        .addConverterFactory(MoshiConverterFactory.create())
+        .build();
+    service = retrofit.create(Service.class);
+  }
+
+  @SuppressWarnings("ConstantConditions") @Test public void githubUserWithOrganization() throws Exception {
+    server.enqueue(mockResponse("src/test/graphql/FelipeGithubUserWithOrganizationsResponse.json"));
+
+    UserWithOrganizationsQuery query = new UserWithOrganizationsQuery(
```

Comment:
is there a reason why the api can't be `UserWithOrganizationsQuery.builder().login("felipecsl") .build()`  The nesting seems extraneous. If I am making a new instance of the Query why have a builder inside of the constructor. I dunno just a thought.